lib/types/src/protocol/proving_version.rs

```diff
@@ -83,8 +89,12 @@ impl ProvingVersion {
         forward_run_execution_version: ExecutionVersion,
     ) -> Self {
         match forward_run_execution_version {
-            ExecutionVersion::V1 | ExecutionVersion::V2 | ExecutionVersion::V3 => Self::V3,
+            // TODO: panic?
+            ExecutionVersion::V1 | ExecutionVersion::V2 | ExecutionVersion::V3 => {
+                panic!("Unsupported execution version")
+            }
```

Comment:
Soft -1 on changing pre-existing logic as a part of bigger PR; makes reviewing harder.
I don't have an opinion here, but I also have no confidence if it can potentially break something in production.

Reply:
Just didn't want to keep more than 2 supported versions of proving, AFAIU, the previous version of proving was needed only for the previous upgrade, now with the new version, we don't need it. cc @perekopskiy